must-gather/bin/gather_knative

```diff
@@ -11,26 +11,34 @@ LOGS_DIR=${LOGS_DIR:-must-gather-logs}
 
 APIRESOURCES=$(${BIN} get crds -o jsonpath="{.items[*].metadata.name}" | tr ' ' '\n' | grep -e knative.dev -e serverless.openshift.io)
 
+COLLECT_NAMESPACES=()
+
 for APIRESOURCE in ${APIRESOURCES[@]}
 do
   NAMESPACES=$(${BIN} get ${APIRESOURCE} --all-namespaces=true -o jsonpath='{range .items[*]}{@.metadata.namespace}{"\n"}{end}' | uniq)
   for NAMESPACE in ${NAMESPACES[@]}
   do
+    COLLECT_NAMESPACES+=("${NAMESPACE}")
     mkdir -p ${LOGS_DIR}/${NAMESPACE}/${APIRESOURCE}
     ${BIN} describe ${APIRESOURCE} -n ${NAMESPACE} > ${LOGS_DIR}/${NAMESPACE}/${APIRESOURCE}/describe.log
     ${BIN} get ${APIRESOURCE} -n ${NAMESPACE} -o=yaml > ${LOGS_DIR}/${NAMESPACE}/${APIRESOURCE}/get.yaml
   done
 done
 
+SYSTEM_NAMESPACES=(knative-eventing knative-serving knative-serving-ingress)
 
-# Collect oc adm inspect and extra resources in knative system namespaces.
+# Collect oc adm inspect in all namespaces with Serverless resources.
+COLLECT_NAMESPACES+=("${SYSTEM_NAMESPACES[@]}")
+for NAMESPACE in $(echo "${COLLECT_NAMESPACES[@]}" | tr " " "\n" | sort --unique)
```

Comment:
From a quick check this can be long (we have -x) and creates noise, but I guess we are ok.

```++ echo knative-serving-ingress knative-serving test1 test2 knative-serving test1 test2 test1 test2 knative-serving test1 test2 test1 test2 test1 test2 test1 test2 test1 test2 test1 test2 knative-eventing knative-serving knative-serving-ingress
```

Reply:
This is the output before running `sort --unique` which will make the list much shorter. Or am I missing something.

Reply:
I was referring to the log output. Imagine having many resources and namespaces.

Reply:
I got this:
```
+ COLLECT_NAMESPACES+=("${SYSTEM_NAMESPACES[@]}")
++ echo knative-serving-ingress knative-serving test knative-serving test test knative-eventing knative-eventing knative-serving test test test test test test knative-eventing knative-serving knative-serving-ingress
++ tr ' ' '\n'
++ sort --unique
+ for NAMESPACE in $(echo "${COLLECT_NAMESPACES[@]}" | tr " " "\n" | sort --unique)
+ oc adm inspect namespace knative-eventing --dest-dir=must-gather-logs
Gathering data for ns/knative-eventing...
Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
Wrote inspect data to must-gather-logs.
+ for NAMESPACE in $(echo "${COLLECT_NAMESPACES[@]}" | tr " " "\n" | sort --unique)
+ oc adm inspect namespace knative-serving --dest-dir=must-gather-logs
Gathering data for ns/knative-serving...
Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
Wrote inspect data to must-gather-logs.
+ for NAMESPACE in $(echo "${COLLECT_NAMESPACES[@]}" | tr " " "\n" | sort --unique)
+ oc adm inspect namespace knative-serving-ingress --dest-dir=must-gather-logs
Gathering data for ns/knative-serving-ingress...
Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
Wrote inspect data to must-gather-logs.
+ for NAMESPACE in $(echo "${COLLECT_NAMESPACES[@]}" | tr " " "\n" | sort --unique)
+ oc adm inspect namespace test --dest-dir=must-gather-logs
Gathering data for ns/test...
Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
Wrote inspect data to must-gather-logs.

```

Reply:
So, it was collecting knative-eventing, knative-serving, knative-serving-ingress and test.

Reply:
Yeah that is fine. I am just saying that this line:
"++ echo knative-serving-ingress knative-serving test knative-serving test test knative-eventing knative-eventing knative-serving test test test test test test knative-eventing knative-serving knative-serving-ingress
"
can be very long :)